### PR TITLE
CVSL-1615 Adding migration endpoint for populating prison numbers

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/MigrationsController.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseBody
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.hmppshdcapi.config.ErrorResponse
+
+const val MIGRATION_ROLE = "HDC_MIGRATION_ADMIN"
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/MigrationsController.kt:20]: could maybe change this to something we already have..

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/PopulatePrisonNumberMigration.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
+package uk.gov.justice.digital.hmpps.hmppshdcapi.licences
+
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.Pageable
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.hmppshdcapi.licences.prison.PrisonSearchApiClient
+import uk.gov.justice.digital.hmpps.hmppshdcapi.licences.prison.Prisoner
+
+const val UNKNOWN_PRISON_NUMBER = "???"
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/PopulatePrisonNumberMigration.kt:10]: idea is to try getting info from prisoner offender search - if we can't find a record in prisoner offender search, we'll mark them with this. 

We could then look at fixing the remaining by direct look up to nomis. This is preferable to battering nomis for all 113,000 licences

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/prison/PrisonApiClient.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import reactor.core.publisher.Mono
+
+@Service
+class PrisonApiClient(@Qualifier("oauthPrisonClient") val prisonerSearchApiWebClient: WebClient) {
+
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun getBooking(bookingId: Long): Booking? {
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/prison/PrisonApiClient.kt:19]: this is not being used at the moment but probably useful in the near future

**Review by louisenorris — APPROVED** [review]
LGTM

**Pull request merged**